File: regions/system/library/python/telemetry/src/k1s0_telemetry/logger.py
"""structlog ベースのロガー設定"""

from __future__ import annotations

import logging
import sys

import structlog


def new_logger(level: str = "INFO", format: str = "json") -> structlog.stdlib.BoundLogger:
    """設定済みの structlog ロガーを返す。

    Args:
        level: ログレベル ("DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL")
        format: 出力形式 ("json" or "text")

    Returns:
        設定済みの structlog.stdlib.BoundLogger
    """
    log_level = getattr(logging, level.upper(), logging.INFO)
    logging.basicConfig(
        format="%(message)s",
        stream=sys.stdout,
        level=log_level,
    )

    processors: list[structlog.types.Processor]
    if format == "json":
        processors = [
            structlog.contextvars.merge_contextvars,
            structlog.processors.add_log_level,
            structlog.processors.TimeStamper(fmt="iso"),
            structlog.processors.StackInfoRenderer(),
            structlog.processors.JSONRenderer(),
        ]
    else:
        processors = [
            structlog.contextvars.merge_contextvars,
            structlog.processors.add_log_level,
            structlog.processors.TimeStamper(fmt="iso"),
            structlog.dev.ConsoleRenderer(),
        ]

    structlog.configure(
        processors=processors,
        wrapper_class=structlog.stdlib.BoundLogger,
        context_class=dict,
        logger_factory=structlog.stdlib.LoggerFactory(),
        cache_logger_on_first_use=True,
    )

    return structlog.stdlib.get_logger()

File: regions/system/library/swift/telemetry/Sources/K1s0Telemetry/Logger.swift
/// 構造化ログ出力（クロスプラットフォーム対応）。
public struct Logger: Sendable {
    private let subsystem: String
    private let category: String

    public init(subsystem: String, category: String) {
        self.subsystem = subsystem
        self.category = category
    }

    public func debug(_ message: String) {
        log(level: "DEBUG", message: message)
    }

    public func info(_ message: String) {
        log(level: "INFO", message: message)
    }

    public func warning(_ message: String) {
        log(level: "WARN", message: message)
    }

    public func error(_ message: String) {
        log(level: "ERROR", message: message)
    }

    private func log(level: String, message: String) {
        print("[\(level)] [\(subsystem)/\(category)] \(message)")
    }
}
